### DISCUSSION TITLE
I try to run the test and it gives me an error and Test ignored

### DISCUSSION THREAD

**Discussion by gonza-mol** [pytest-dev/pytest#12127] [Q&A]
I have a framework with selenium and python (using pycharm), the issue was that it worked well for me, and I had to format the PC, and reinstall pycharm, python and everything necessary. And when I want to run the different projects again, it now gives me the following error and none of my tests work in any way, neither using the commands nor the interface. I have already changed all kinds of settings and I have not been able to find the error. Could you tell me if there is a solution, or if you know the problem?
Python: 3.8.0
pytest 4.1.1
![BaseClass](https://github.com/pytest-dev/pytest/assets/86359451/266fac31-7820-452d-9676-9d22303bca59)
![conftestFile](https://github.com/pytest-dev/pytest/assets/86359451/a2688947-d085-49b4-a1fe-3ff1aa8c3469)
![error on the logs](https://github.com/pytest-dev/pytest/assets/86359451/0891d58a-c00a-421c-b82e-99097d1c53fa)
![python interpreter configuration](https://github.com/pytest-dev/pytest/assets/86359451/e1039d12-52a1-4dd1-aa41-d6d9ea053503)
![structure of framework](https://github.com/pytest-dev/pytest/assets/86359451/690de5ec-d27b-4a19-8f9c-c7aef3da216d)
![test](https://github.com/pytest-dev/pytest/assets/86359451/899ef73f-638f-4475-99d1-1f511cc94c63)
![BaseClass](https://github.com/pytest-dev/pytest/assets/86359451/8c2d2daf-9028-45c6-bdaa-75e4dcd4ded1)
![conftestFile](https://github.com/pytest-dev/pytest/assets/86359451/85500bb3-e1bd-4891-8ef5-9ee26db43742)
![error on the logs](https://github.com/pytest-dev/pytest/assets/86359451/d12ffa9b-3782-40c7-80bd-60fa8ae4d280)
![python interpreter configuration](https://github.com/pytest-dev/pytest/assets/86359451/8744fb4a-7f74-448f-a1f8-3282e9a8d4ca)
![structure of framework](https://github.com/pytest-dev/pytest/assets/86359451/514833d1-37c9-446c-842b-69788f098b20)
![test](https://github.com/pytest-dev/pytest/assets/86359451/aa00ab67-4beb-4274-b6c1-474ba48a4cc8)

**Comment by RonnyPfannschmidt** [comment]
This looks like a user error, driver is none in the local conftest

Please copy and paste text, screenshots of code and tracebacks are very rude

**Comment by gonzamol17** [comment]
Hey Ronny, yes sure:
This is the code of one from my test:

```
import time
import pytest
import unittest
import sys
import os
sys.path.append(os.path.join(os.path.dirname(__file__),"..",".."))
import HtmlTestRunner
from Utils import Utils as Utils
from Utils.BaseClass import BaseClass
from POM.HomePage import HomePage
from POM.AccordionsPage import AccordionsPage


@pytest.mark.usefixtures("test_setup")
class TestAccordions(BaseClass):

    def test_Accordions(self):
        log = self.get_Logger()
        driver = self.driver
        hp = HomePage(driver)
        time.sleep(1)
        hp.closeCookiesWindows()
        time.sleep(1)
        driver.execute_script("window.scrollTo(0, 300)")
        time.sleep(1)
        hp.clickBtnAccordions()
        ap = AccordionsPage(driver)
        time.sleep(1)
        ap.selectAccordion()
        time.sleep(1)
        assert ap.getTitleFromAccordion() == "This is an accordion item."
        time.sleep(2)
```

--------------------------------------------------------------------

This is the code of conftest

```
import time
import warnings
import json
import pytest
from Utils import Utils
from selenium.common.exceptions import NoSuchElementException
driver = None

def pytest_addoption(parser):
    parser.addoption("--browser", action="store", default="chrome")

@pytest.fixture()
def test_setup(request):
    global driver
    from selenium import webdriver
    browser = request.config.getoption("--browser")
    if browser == 'chrome':
        driver = webdriver.Chrome("C:\\Users\\admin\\PycharmProjects\\Practice-Automation\\Drivers\\chromedriver.exe")
    elif browser == 'firefox':
        driver = webdriver.Firefox("C:\\Users\\admin\\PycharmProjects\\Practice-Automation\\Drivers\\geckodriver.exe")
    warnings.simplefilter('ignore', ResourceWarning)
    #driver = webdriver.Chrome("C:\\Users\\admin\\PycharmProjects\\SegundoProyecto\\Drivers\\chromedriver.exe")
    driver.implicitly_wait(10)
    driver.maximize_window()
    request.cls.driver = driver
    driver.get(Utils.URL)
    #time.sleep(3)
    #driver.switch_to.frame("framelive")
    yield
    driver.close()
    driver.quit()




@pytest.mark.hookwrapper
def pytest_runtest_makereport(item):
    """
        Extends the PyTest Plugin to take and embed screenshot in html report, whenever test fails.
        :param item:
        """
    pytest_html = item.config.pluginmanager.getplugin('html')
    outcome = yield
    report = outcome.get_result()
    extra = getattr(report, 'extra', [])

    if report.when == 'call' or report.when == "setup":
        xfail = hasattr(report, 'wasxfail')
        if (report.skipped and xfail) or (report.failed and not xfail):
            file_name = report.nodeid.replace("::", "_") + ".png"
            _capture_screenshot(file_name)
            if file_name:
                html = '<div><img src="%s" alt="screenshot" style="width:304px;height:228px;" ' \
                       'onclick="window.open(this.src)" align="right"/></div>' % file_name
                extra.append(pytest_html.extras.html(html))
        report.extra = extra


def _capture_screenshot(name):
    #driver.get_screenshot_as_file(name)
    driver.get_screenshot_as_file("C:\\Users\\admin\\PycharmProjects\\Practice-Automation\\Screenshots\\"+name)
```
-----------------------------------------------------------------------------------

this is the content from the BaseClass:

```
import inspect
import pytest
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import logging

#ver si en esta clase base esta anotación de pytest.mark va también, porque ya la estamos usando en la clase
# que contiene el test. Por lo tanto dentro de la clase del Test, lo único que hay que hacer es llamar
# a esta clase base, para que pueda usar estos métodos heredados
#@pytest.mark.usefixtures("test_setup")
class BaseClass:

    def get_Logger(self):
        loggername = inspect.stack()[1][3]
        logger = logging.getLogger(loggername)
        fileHandler = logging.FileHandler("C:\\Users\\admin\\PycharmProjects\\Practice-Automation\\Data\\logfile.log")
        formatter = logging.Formatter("%(asctime)s :%(levelname)s : %(name)s :%(message)s")
        fileHandler.setFormatter(formatter)
        logger.addHandler(fileHandler)
        #logger = logging.FileHandler(filemode='w')
        logger.setLevel(logging.DEBUG)
        return logger
```
-----------------------------------------------------------------------------------------

and the file from the utils file, where i alocate the url:

```
URL = "https://practice-automation.com/"
```
-----------------------------------------------------------------------------------------------------------------------

And one of the files that i am used on the Page Object Model for this object:

```
import time

from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as ec


class AccordionsLocators:
    arrowAccordion = (By.CSS_SELECTOR, "#post-1261 > div summary")
    titleFromAccordion = (By.CSS_SELECTOR, "#post-1261 details > div > p")


class AccordionsPage:

    def __init__(self, driver):
        self.driver = driver

    def selectAccordion(self):
        return self.driver.find_element(*AccordionsLocators.arrowAccordion).click()

    def getTitleFromAccordion(self):
        return self.driver.find_element(*AccordionsLocators.titleFromAccordion).text
```

**Comment by The-Compiler** [comment]
I converted this to a discussion, as this doesn't look like an issue with pytest. Please edit your post to use [proper code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks), right now it's pretty much unreadable.

**Reply by gonzamol17** [reply]
Thanks for the clarifications and tips.

**Comment by The-Compiler** [comment]
If your `test_setup` fixture fails to create `driver`, the `driver` global is not set (i.e. `None`), but in your `pytest_runtest_makereport` you then call `_capture_screenshot` which accesses that global.

**Comment by Jason1roy** [comment]
It sounds like you may be encountering issues related to environment setup or dependencies. Here are some steps you can take to troubleshoot and resolve the problem:

Check Python Interpreter: Ensure that PyCharm is using the correct Python interpreter. Go to File > Settings > Project > Python Interpreter and verify that the interpreter path is pointing to the correct Python executable.

Install Dependencies: Make sure that all necessary dependencies are installed in your Python environment. You can use pip to install required packages by running pip install <package-name> in the terminal or using the package manager in PyCharm.

Check Project Configuration: Verify that your project configuration is set up correctly. Ensure that the project structure is configured properly and that all necessary files and directories are included.

Check Selenium WebDriver: If you are using Selenium WebDriver, ensure that the appropriate driver executable (e.g., chromedriver for Chrome, geckodriver for Firefox) is installed and available in your system PATH or specified in your project settings.

Review Code Changes: If you made any changes to your code or project structure before formatting your PC, review those changes to see if they might be causing the issue. Check for any syntax errors, import errors, or other issues in your code.

Check System Environment: Verify that your system environment is configured correctly. Ensure that there are no conflicting installations of Python or other software that could be causing issues.

Update PyCharm: Make sure you are using the latest version of PyCharm. You can check for updates by going to Help > Check for Updates.

Consult Documentation: Refer to the documentation for PyCharm, Python, and Selenium for troubleshooting tips and guidance. Look for any known issues or compatibility issues that may be relevant to your setup.

**Reply by The-Compiler** [reply]
Just copy-pasting ChatGPT answers that have nothing to do with the issue at hand isn't helpful.
